UKHO.PeriodicOutputService/UKHO.PeriodicOutputService.Common/Services/FssService.cs

```diff
@@ -375,25 +376,85 @@ public async Task<IEnumerable<BatchFile>> GetAioInfoFolderFilesAsync(string batc
+                    throw new FulfilmentException(EventIds.ReadMeTextFileNotFound.ToEventId());
+                }
+            }
+            else
+            {
+                _logger.LogError(EventIds.QueryFileShareServiceReadMeFileNonOkResponse.ToEventId(), "Error in file share service while searching readme.txt file with uri {RequestUri} responded with {StatusCode} for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", httpResponse.RequestMessage.RequestUri, httpResponse.StatusCode, batchId, correlationId);
+                throw new FulfilmentException(EventIds.QueryFileShareServiceReadMeFileNonOkResponse.ToEventId());
+            }
+        }
+
+        public async Task<bool> DownloadReadMeFileAsync(string readMeFilePath, string batchId, string exchangeSetRootPath, string correlationId)
+        {
+            var accessToken = await _authFssTokenProvider.GetManagedIdentityAuthAsync(_fssApiConfiguration.Value.FssClientId);
+            string fileName = _fssApiConfiguration.Value.ReadMeFileName;
+            string filePath = Path.Combine(exchangeSetRootPath, fileName);
+
+            HttpResponseMessage httpReadMeFileResponse = await _fssApiClient.DownloadFile(readMeFilePath.TrimStart('/'), accessToken);
+
+            var requestUri = new Uri(httpReadMeFileResponse.RequestMessage.RequestUri.ToString()).GetLeftPart(UriPartial.Path);
+
+            if (httpReadMeFileResponse.IsSuccessStatusCode)
+            {
+                var serverValue = httpReadMeFileResponse.Headers.Server.ToString().Split('/').First();
+                await using Stream stream = await httpReadMeFileResponse.Content.ReadAsStreamAsync();
+                if (serverValue == ServerHeaderValue)
+                {
+                    _logger.LogInformation(EventIds.DownloadReadmeFile307RedirectResponse.ToEventId(), "File share service download readme.txt file redirected with uri:{requestUri} responded with 307 code for BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", requestUri, batchId, correlationId);
+                }
+                return _fileSystemHelper.DownloadReadmeFile(filePath, stream);
+            }
+            else
+            {
+                _logger.LogError(EventIds.DownloadReadMeFileNonOkResponse.ToEventId(), "Error in file share service while downloading readme.txt file with uri:{requestUri} responded with {StatusCode} and BatchId:{BatchId} and _X-Correlation-ID:{CorrelationId}", requestUri, httpReadMeFileResponse.StatusCode, batchId, correlationId);
+                throw new FulfilmentException(EventIds.DownloadReadMeFileNonOkResponse.ToEventId());
+            }
+        }
+
         //Private Methods
         [ExcludeFromCodeCoverage]
         private CreateBatchRequestModel CreateBatchRequestModel(Batch batchType)
```

Comment:
Perhaps not a change for this PR but something to keep in mind, this method is 127 lines long! We should try split these methods out into smaller parts

Reply:
Sure, Dan. we have PBI for Builder service upliftment we will try to split and refactor this method.